src/components/MoviesList/MoviesList.jsx

```diff
@@ -1,23 +1,38 @@
-import React, { useEffect } from "react";
+import React, {useEffect} from "react";
 import "./movies-list.scss";
 import MovieBox from "./MovieBox/MovieBox";
-import { connect } from "react-redux";
-import { bindActionCreators } from "redux"
-import {getMoviesList} from "../../store/movie/action-creators";
+import {connect} from "react-redux";
+import {useParams} from 'react-router-dom';
+import {searchMovies} from "../../store/movie/action-creators";
+
 
 const MoviesList = (props) => {
+    const {searchQuery} = useParams();
 
     useEffect(() => {
-        props.getMovies();
-    }, []);
+        if (searchQuery) {
+            props.searchMovies(searchQuery, "title");
+        }
+    }, [searchQuery]);
 
-    const {moviesList, totalAmount} = props;
+    const {moviesList} = props;
 
-    return(
-        <div className="movies-list">
-            {moviesList.map( movie => <MovieBox key={`movie-box-${movie.id}`} movie={movie} /> ) }
-        </div>
-    )
+    if (!moviesList.length) {
+        return (
+            <div className="movies-container">
+                <h2>No movie found</h2>
+            </div>
+        )
+    } else {
```

Comment:
minor: we actually can avoid else statement here since we have return keyword in `if`. it would be a little bit cleaner.
just a thought, no action required